newIDE/app/src/SceneEditor/index.js

```diff
@@ -212,6 +215,9 @@ export default class SceneEditor extends React.Component<Props, State> {
       },
 
       selectedObjectTags: [],
+
+      renamedObjectWithContext: null,
+      selectedObjectWithContext: null,
```

Comment:
We now unfortunately have two things:
`selectedObjectNames` and `selectedObjectWithContext`.

Which one is the source of truth? When do one is used and not the other? This is not clear because the good answer is: "this should be a single variable and not two".

This is not an easy task, but we need to replace one with the other. It's important because otherwise we start introducing two things doing the same thing, which means bugs and a hard to understand code for future developers.
I *think* we should replace `selectedObjectNames` by `selectedObjectsWithContext` (which would be an array of objects with contexts). 

Note that I'm maybe mistaken, but if that's the case, the names of the variables should be changed to make clear what is used for what :)

Reply:
When I began the task I thought the same but then I saw that I had to change too many lines and perhaps I was not allowed to do so many changes.

If you agree, I could remove the `selectedObjectNames` variable, that is only a array of strings, and use only `selectedObjectsWithContext` (or `selectedObjects`), that has more information.